Tests/WebUITests/WebViewProxyTests.swift

```diff
@@ -87,3 +210,67 @@ struct WebViewProxyTests {
         #expect(oldInstance != newInstance)
     }
 }
+
+private func wait<V: Equatable & Sendable>(
+    for sequence: AsyncPublisher<Published<V?>.Publisher>,
+    condition: @escaping @Sendable (V?) -> Bool,
+    timeout: Duration
+) async throws -> Bool {
+    try await withThrowingTaskGroup(of: Bool.self) { group in
+        defer { group.cancelAll() }
+
+        group.addTask {
+            try await Task { @MainActor in
+                for try await value in sequence {
+                    if condition(value) {
+                        return true
+                    }
+                }
+                return false
+            }
+            .value
+        }
+
+        group.addTask {
+            try await Task.sleep(for: timeout)
+            return false
+        }
+
+        guard let result = try await group.next() else {
+            return false
+        }
+        return result
+    }
+}
+
+private func wait<V: Equatable & Sendable>(
+    for sequence: AsyncPublisher<Published<V>.Publisher>,
+    condition: @escaping @Sendable (V) -> Bool,
+    timeout: Duration
+) async throws -> Bool {
+    try await withThrowingTaskGroup(of: Bool.self) { group in
+        defer { group.cancelAll() }
+
+        group.addTask {
+            try await Task { @MainActor in
+                for try await value in sequence {
+                    if condition(value) {
+                        return true
+                    }
+                }
+                return false
+            }
+            .value
```

Comment:
It looks like this part could be refactored as follows:
```suggestion
            await sequence.first(where: { condition($0) }) != nil
```

Reply:
I had no idea about writing it this way.

Reply:
Is `MainActor` not necessary to access the `sequence`?

Reply:
Changed in [b573db9](https://github.com/cybozu/WebUI/pull/59/commits/b573db930887fdd02465869f95936856d6f5da85)